### PR TITLE
DEVPROD-6200 Delete unused distro routes

### PR DESCRIPTION
**Pull request by malikchaya2**
DEVPROD-6200

### Description
According to splunk the routes are unused.

### CODE REVIEW

File: rest/route/distro.go [hunk 1]
```diff
@@ -742,182 +579,6 @@ func validateDistro(ctx context.Context, apiDistro *model.APIDistro, resourceID
 	return d, nil
 }
 
-///////////////////////////////////////////////////////////////////////
-//
-// PATCH /rest/v2/distros/{distro}/execute
-
-type distroExecuteHandler struct {
```
Comment (Kimchelly) [inline rest/route/distro.go:749]: The CLI and [REST client](https://github.com/evergreen-ci/evergreen/blob/8db3a5c7fe497090c13f0411d2881eda5cc372eb/rest/client/methods.go#L678-L704) logic to use this route also needs to be deleted.

File: rest/route/distro.go [hunk 2]
```diff
@@ -742,182 +579,6 @@ func validateDistro(ctx context.Context, apiDistro *model.APIDistro, resourceID
...
-	opts   model.APIDistroScriptOptions
```
Comment (Kimchelly) [inline rest/route/distro.go:750]: This struct can be deleted if the CLI logic is deleted.

**Review by ablack12 — APPROVED** [review]

File: rest/route/distro.go [hunk 3]
```diff
@@ -502,164 +497,6 @@ func (h *distroAMIHandler) Run(ctx context.Context) gimlet.Responder {
 	return gimlet.MakeJSONErrorResponder(errors.Errorf("no settings available for region '%s' for distro '%s'", h.region, h.distroID))
 }
 
-////////////////////////////////////////////////////////////////////////
-//
-// PATCH /rest/v2/distros/settings modifies provider settings across all distros for the given region
```
Comment (Kimchelly) [inline rest/route/distro.go:507]: The unit tests need to be deleted as well.

File: rest/route/distro.go [hunk 4]
```diff
@@ -742,182 +579,6 @@ func validateDistro(ctx context.Context, apiDistro *model.APIDistro, resourceID
 	return d, nil
 }
 
-///////////////////////////////////////////////////////////////////////
-//
-// PATCH /rest/v2/distros/{distro}/execute
```
Comment (Kimchelly) [inline rest/route/distro.go:747]: The unit tests need to be deleted as well.

**Review by Kimchelly — CHANGES_REQUESTED** [review]

File: rest/route/distro.go (hunk 1 above)
Comment (malikchaya2) [reply rest/route/distro.go:749]: done

File: rest/route/distro.go (hunk 2 above)
Comment (malikchaya2) [reply rest/route/distro.go:750]: done

**Review by Kimchelly — APPROVED** [review]
LGTM mod `APIDistroScriptOptions` still has to be deleted from the REST models.

**Pull request merged**